internal/command/command.go

```diff
@@ -355,10 +356,246 @@ var CmdUpdateApis = &Command{
+			if flagSkipBuild {
+				if err := container.Build(flagImage, "repo-root", repoPath, "library-id", library.Id); err != nil {
+					slog.Info(fmt.Sprintf("Received error running container: '%s'", err))
+					//TODO: log in release PR
+				}
+			}
+
+			//TODO: add extra meta data what is this
+			prDescription += fmt.Sprintf("releasing lib: %s:%s\n", library.Id, library.NextVersion)
+			librariesToRelease[library.Id] = releaseVersion
+
+			createCommit(ctx, err, repo, releaseNotes)
+
+			//TODO: refactor, see if can wipe on every library instead of deleting
+			err = os.Remove(filepath.Join(inputDirectory, "release-notes.txt"))
+			if err != nil {
+				return nil, "", err
+			}
+		}
+	}
+	return librariesToRelease, prDescription, nil
+}
+
+func createCommit(ctx context.Context, err error, repo *gitrepo.Repo, releaseNotes string) {
+	_, err = gitrepo.AddAll(ctx, repo)
+	if err != nil {
+		fmt.Println("Error adding files:", err)
+		//TODO update PR description with this data and mark as not humanly resolvable
+	}
+	//TODO: what should the description be here?
+	if err := gitrepo.Commit(ctx, repo, releaseNotes); err != nil {
+		slog.Info(fmt.Sprintf("Received error trying to commit: '%s'", err))
+		//TODO update PR description with this data and mark as not humanly resolvable
+	}
+}
+
+// TODO: update with actual logic
+func createReleaseNotes(library *statepb.LibraryReleaseState, commitMessages []string, inputDirectory string) (string, error) {
+	releaseNotes := fmt.Sprintf("Library: %s\n", library.Id)
+
+	for _, commitMessage := range commitMessages {
+		releaseNotes += fmt.Sprintf("%s\n", commitMessage)
+	}
+
+	path := filepath.Join(inputDirectory, "release-notes.txt")
+	file, err := os.Create(path)
+	if err != nil {
+		fmt.Println("Error creating release notes file", err)
+		return "", err
+	}
+	_, err = file.WriteString(releaseNotes)
+	if err != nil {
+		return "", err
+	}
+	return releaseNotes, nil
+}
+
+// TODO: need to add handling of suffix
+func calculateNextVersion(library *statepb.LibraryReleaseState) (string, error) {
+	if library.NextVersion != "" {
+		return library.NextVersion, nil
+	}
+	parts := strings.Split(library.CurrentVersion, ".")
+	if len(parts) != 3 {
+		return "", fmt.Errorf("invalid version format: %s", library.CurrentVersion)
+	}
+
+	major, err := strconv.Atoi(parts[0])
+	if err != nil {
+		return "", fmt.Errorf("invalid major version: %w", err)
+	}
+
+	minor, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return "", fmt.Errorf("invalid minor version: %w", err)
+	}
+
+	suffix := strings.Split(parts[2], "-")
+	patch := "0"
+	if len(suffix) > 1 {
+		patch += "-" + suffix[1]
+	}
+
+	//increment minor version
+	minor++
+
+	return fmt.Sprintf("%d.%d.%s", major, minor, patch), nil
+}
+
+func updateLibraryMetadata(releasedLibraries map[string]string, pipelineState *statepb.PipelineState, repoPath string) {
+	for i := 0; i < len(pipelineState.LibraryReleaseStates); i++ {
+		library := pipelineState.LibraryReleaseStates[i]
```

Comment:
As noted before, let's do this within generateReleaseCommitForEachLibrary. We should also copy the "last generated commit" for each API into the library's metadata.